src/main/java/net/neoforged/neoforge/common/extensions/IItemStackExtension.java

```diff
@@ -75,7 +76,8 @@ default boolean hasCraftingRemainingItem() {
      *         decide.
      */
     default int getBurnTime(@Nullable RecipeType<?> recipeType) {
-        return self().getItem().getBurnTime(self(), recipeType);
+        var fuel = self().getItemHolder().getData(NeoForgeDataMaps.FURNACE_FUELS);
```

Comment:
Shouldn't this still be checking if the burn time is overridden on the item to allow for nbt sensitive fuels?